dfastmi/batch/ConfigurationCheckerLegacy.py

```diff
@@ -0,0 +1,426 @@
+# -*- coding: utf-8 -*-
+"""
+Copyright (C) 2024 Stichting Deltares.
+
+This library is free software; you can redistribute it and/or
+modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation version 2.1.
+
+This library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with this library; if not, see <http://www.gnu.org/licenses/>.
+
+contact: delft3d.support@deltares.nl
+Stichting Deltares
+P.O. Box 177
+2600 MH Delft, The Netherlands
+
+All indications and logos of, and references to, "Delft3D" and "Deltares"
+are registered trademarks of Stichting Deltares, and remain the property of
+Stichting Deltares. All rights reserved.
+
+INFORMATION
+This file is part of D-FAST Morphological Impact: https://github.com/Deltares/D-FAST_Morphological_Impact
+"""
+import configparser
+from typing import Callable, List, Optional, Tuple
+from dfastmi.batch.AConfigurationChecker import AConfigurationCheckerBase
+import dfastmi.kernel.core
+from dfastmi.io.Reach import ReachLegacy
+from dfastmi.io.RiversObject import RiversObject
+from dfastmi.kernel.core import BoolVector, QRuns, Vector
+
+WAQUA_EXPORT = "WAQUA export"
+DFLOWFM_MAP = "D-Flow FM map"
+
+class ConfigurationCheckerLegacy(AConfigurationCheckerBase):
+    """
+        Check if a version 1 / legacy analysis configuration is valid.
+    """
+    _validators = {}
+    """Contains the configuration validator be used depending on mode"""
+
+    def __init__(self):
+        self._register_validator(WAQUA_EXPORT, self._check_configuration_cond_waqua)
+        self._register_validator(DFLOWFM_MAP, self._check_configuration_cond_fm)
+
+    def _register_validator(self, mode: str, validator: Callable[[configparser.ConfigParser,  int], bool]):
```

Comment:
maybe move this to a seperate class or a seperate private class if it is only used here

Reply:
Ralph will think about it.

Reply:
Created separate class